File: VB_Lib/Extensions/Convert.vb
Public Class Convert
    Enum Languages
        vbnet = 1
        CSharp = 2
        CPlusPlus = 3
    End Enum
    Public Shared Sub WriteHelpExtension(ByVal Name As String, 
										 ByVal PixSource As Pix, 
										 ByVal PixTarget As Pix, 
										 ByVal Line1 As String(), 
										 Optional Lang As Languages = Languages.vbnet)
										 
        If Not IsNothing(PixSource) Then PixSource.save_jpg("..\..\ALL_Help\HelpImages\" &  Name & "_S.jpg")
        If Not IsNothing(PixTarget) Then PixTarget.save_jpg("..\..\ALL_Help\HelpImages\" & Name & "_T.jpg")

        If My.Computer.FileSystem.FileExists("..\..\VB_Lib\IncludeComments.xml") = True Then
            Dim XML As New Xml.XmlDocument : XML.Load("..\..\VB_LIB\IncludeComments.xml")
            Dim Node As Xml.XmlNode = XML.SelectSingleNode("Comments/" & Name) : If IsNothing(Node) Then Exit Sub
            Dim CodeNode = Node.SelectSingleNode("example/code[@language=""" & Lang.ToString & """]") : If IsNothing(CodeNode) Then Exit Sub
            CodeNode.InnerXml = vbCrLf & String.Join(vbCrLf, Line1)
            XML.Save("IncludeComments.xml")
        End If
    End Sub
    Public Shared Function Deg2Grad(ByVal Deg As Double) As Double
        Return Deg * (Math.PI / 180)
    End Function
    Public Shared Function Grad2Dec(ByVal Grad As Double) As Double
        Return Grad / (Math.PI / 180)
    End Function
End Class

File: CS_LIB/Extensions/Convert.cs
using System;
using System.IO ;
using System.Collections;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;

public class Convert
{
	public enum Languages {
		vbnet = 1,
		CSharp = 2,
		CPlusPlus = 3
	}
	public static void WriteHelpExtension(string Name, Pix PixSource, Pix PixTarget, string[] Line1, Languages Lang = Languages.vbnet)
	{
		if (PixSource != null)
		{ PixSource.pixWriteJpeg ("..\\..\\ALL_Help\\HelpImages\\" + Name + "_S.jpg", 75, 0); }

		if (PixTarget != null)
		{ PixTarget.pixWriteJpeg("..\\..\\ALL_Help\\HelpImages\\" + Name + "_T.jpg", 75, 0); }

		if (File.Exists("..\\..\\VB_Lib\\IncludeComments.xml") == true) {
			System.Xml.XmlDocument XML = new System.Xml.XmlDocument();
			XML.Load("..\\..\\VB_LIB\\IncludeComments.xml");
			System.Xml.XmlNode Node = XML.SelectSingleNode("Comments/" + Name);

			if ((Node == null))
			{ return; }

			var CodeNode = Node.SelectSingleNode("example/code[@language=\"" + Lang.ToString() + "\"]");

			if ((CodeNode == null))
			{ return; }

			CodeNode.InnerXml = Environment.NewLine + string.Join(Environment.NewLine, Line1);
			XML.Save("IncludeComments.xml");
		}
	}
	public static double Deg2Grad(double Deg)
	{
		return Deg * (Math.PI / 180);
	}
	public static double Grad2Dec(double Grad)
	{
		return Grad / (Math.PI / 180);
	}
}

